run.sh

```diff
@@ -1,34 +1,71 @@
 #!/usr/bin/env bash
 set -e -o pipefail
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+CLEAR='\033[0m'
 
-if [ -z "$1" ]
-  then
-    TENV_REGTEST="trezor-user-env-regtest"
-elif [[ $1 == --no-regtest ]]; then
-    TENV_REGTEST=""
-fi
-
+TENV_REGTEST="trezor-user-env-regtest"
 SYSTEM=$(uname)
+XHOST_ADDRESS=""
+SERVICE_NAME=""
+PULL=1
+BRANCH=$(git rev-parse --abbrev-ref HEAD)
 
-if [[ $SYSTEM == Linux* ]]; then
+function usage() {
+  if [ -n "$1" ]; then
+    echo -e "$1";
+  fi
+  echo "Usage: $0 [-r --no-regtest] [-p --no-pull]"
+  echo "  -r, --no-regtest      Run tenv only, no regtest"
+  echo "  -p, --no-pull         Do not pull changes, neither git nor docker. Automatically set to True when not on 'master'"
+  echo ""
+  exit 1
+}
 
-    echo -e "Setup xhost for video device output"
-    xhost +
-    echo -e "Downloading latest images"
-    docker-compose -f ./docker/compose.yml pull trezor-user-env-unix $TENV_REGTEST
-    echo -e "Starting trezor-user-env"
-    docker-compose -f ./docker/compose.yml up trezor-user-env-unix $TENV_REGTEST
+# parse params
+while [[ "$#" > 0 ]]; do case $1 in
+  -r|--no-regtest) TENV_REGTEST=""; shift;;
+  -p|--no-pull) PULL=0; shift;;
+  -h|--help) usage; shift;;
+  *) usage "${RED}Unknown parameter passed: $1${CLEAR}\n"; shift;;
+esac; done
 
-elif [[ $SYSTEM == Darwin* ]]; then
+if [[ $SYSTEM == Linux* ]]; then
+    SERVICE_NAME="trezor-user-env-unix"
+    OPEN="xdg-open"
 
-    echo -e "Setup xhost for video device output"
-    xhost +127.0.0.1
-    echo -e "Downloading latest images"
-    docker-compose -f ./docker/compose.yml pull trezor-user-env-mac $TENV_REGTEST
-    echo -e "Starting trezor-user-env"
-    docker-compose -f ./docker/compose.yml up trezor-user-env-mac $TENV_REGTEST
+elif [[ $SYSTEM == Darwin* ]]; then
+    XHOST_ADDRESS="127.0.0.1"
+    SERVICE_NAME="trezor-user-env-mac"
+    OPEN="open"
 
 else
-    echo -e "Not a supported system - $SYSTEM"
+    echo -e "${RED}Not a supported system: $SYSTEM${CLEAR}\n"
     exit 1
 fi
+
+echo -e "On system: $SYSTEM"
+
+if [[ $PULL -eq 1 ]]; then
+    if [[ $BRANCH != "master" ]]; then
+        echo -e "${RED}Not on master, not pulling latest changes${CLEAR}\n"
+    else
+        echo -e "Pulling git changes"
+        git pull
+
+        echo -e "Downloading latest images"
+        docker-compose -f ./docker/compose.yml pull $SERVICE_NAME $TENV_REGTEST
+    fi
+fi
+
+echo -e "Setup xhost for video device output"
+xhost "+$XHOST_ADDRESS"
+
+if [[ $TENV_REGTEST == "trezor-user-env-regtest" ]]; then
+    echo -e "${GREEN}Starting trezor-user-env with Bitcoin regtest${CLEAR}\n"
+else
+    echo -e "${GREEN}Starting trezor-user-env${CLEAR}\n"
+fi
+(sleep 10 && $OPEN "http://localhost:9002")&
```

Comment:
Not really sure if this is a good idea as it is not connected to the result of `docker-compose`. I am bit worried it will confuse users when something fails.

Reply:
Not sure either, but in most cases, it should ve alright. In the worst case, the browser will say `Unable to connect`. We could theoretically `ping` that URL and only open it in the browser if it suceeds

Reply:
This worked for me - `curl http://localhost:9002 -s -o /dev/null && xdg-open http://localhost:9002`, only opening the dashboard if it runs properly

Reply:
I tried a similar approach with netstat and grep which did not work for me, not sure why. Thanks, let me try this one.

Reply:
Seems like this works! 

`(while ! $(curl http://localhost:9002 -s -o /dev/null); do sleep 1; done; $OPEN "http://localhost:9002")&`

Reply:
RIght, I was about to comment with almost the same one which worked for me :D `while ! curl http://localhost:9002 -s -o /dev/null ; do sleep 0.5; done && xdg-open http://localhost:9002`

Reply:
8d3e758dbc0d7c16c32f0008c9424644f5625676

Reply:
Nice, I verified it works fine in `run.sh`. I just think the fixup points to a bad commit, it should not be to `simplify README` :)

Reply:
Also, theoretically, we are using `http://localhost:9002` more than once, so it might be useful to create `DASHBOARD_URL` variable

Reply:
> Nice, I verified it works fine in run.sh. I just think the fixup points to a bad commit, it should not be to simplify README :)

I have accidentaly added the README commit 🤦. Let me fix it.

Reply:
> Also, theoretically, we are using http://localhost:9002 more than once, so it might be useful to create DASHBOARD_URL variable

👍 79a2a07fd90116fac44f8f313f1c80ee929e2b10